src/app/zap-templates/zcl/data-model/chip/device-energy-management-cluster.xml

```diff
@@ -0,0 +1,196 @@
+      <access op="invoke" privilege="operate"/>
+      <description>Allows a client to cancel the PauseRequest command and enable earlier resumption of operation.</description>
+    </command>
+    <command source="client" code="0x0005" name="ModifyForecastRequest" optional="true" apiMaturity="provisional">
+      <access op="invoke" privilege="operate"/>
+      <arg name="ForecastId" type="int32u"/>
+      <arg name="SlotAdjustments" array="true" type="SlotAdjustmentStruct"/>
+      <description>Allows a client to modify a Forecast within the limits allowed by the ESA.</description>
+    </command>
+    <command source="client" code="0x0006" name="RequestConstraintBasedForecast" optional="true" apiMaturity="provisional">
+      <access op="invoke" privilege="operate"/>
+      <arg name="Constraints" array="true" type="ConstraintsStruct"/>
+      <description>Allows a client to ask the ESA to recompute its Forecast based on power and time constraints.</description>
+    </command>
+    <event code="0x0000" name="PowerAdjustStart" priority="info" side="server" apiMaturity="provisional" optional="true">
+      <description>PowerAdjustStart</description>
+    </event>
+    <event code="0x0001" name="PowerAdjustEnd" priority="info" side="server" apiMaturity="provisional" optional="true">
+      <description>PowerAdjustEnd</description>
+      <field id="0" name="Cause" type="CauseEnum" apiMaturity="provisional"/>
+      <field id="1" name="Duration" type="elapsed_s" apiMaturity="provisional"/>
+      <field id="2" name="EnergyUse" type="int64s" apiMaturity="provisional"/>
+    </event>
+    <event code="0x0002" name="Paused" priority="info" side="server" apiMaturity="provisional" optional="true">
+      <description>Paused</description>
+    </event>
+    <event code="0x0003" name="Resumed" priority="info" side="server" apiMaturity="provisional" optional="true">
+      <description>Resumed</description>
+    </event>
+  </cluster>
+  <enum name="CostTypeEnum" type="enum8" apiMaturity="provisional">
+    <cluster code="0x0098"/>
+    <item name="Financial" value="0x00"/>
+    <item name="GHGEmissions" value="0x01"/>
+    <item name="Comfort" value="0x02"/>
+    <item name="Temperature" value="0x03"/>
+  </enum>
+  <enum name="ESATypeEnum" type="enum8" apiMaturity="provisional">
+    <cluster code="0x0098"/>
+    <item name="EVSE" value="0x00"/>
+    <item name="SpaceHeating" value="0x01"/>
+    <item name="WaterHeating" value="0x02"/>
+    <item name="SpaceCooling" value="0x03"/>
+    <item name="SpaceHeatingCooling" value="0x04"/>
+    <item name="BatteryStorage" value="0x05"/>
+    <item name="SolarPV" value="0x06"/>
+    <item name="FridgeFreezer" value="0x07"/>
+    <item name="WashingMachine" value="0x08"/>
+    <item name="Dishwasher" value="0x09"/>
+    <item name="Cooking" value="0x0A"/>
+    <item name="HomeWaterPump" value="0x0B"/>
+    <item name="IrrigationWaterPump" value="0x0C"/>
+    <item name="PoolPump" value="0x0D"/>
```

Comment:
I couldn't find these in current spec.

Reply:
See note above about this being autogen'd from the in flight Spec PR.